apisix/plugins/response-rewrite.lua

```diff
@@ -87,13 +112,26 @@ function _M.check_schema(conf)
         end
     end
 
+    if conf.vars then
+        local ok, err = expr.new(conf.vars)
+        if not ok then
+            return nil, "failed to validate the 'vars' expression: " .. err
+        end
+    end
+
     return true
 end
 
 
 do
 
 function _M.body_filter(conf, ctx)
+
+    local ok =  vars_matched(conf, ctx)
```

Comment:
We should cache the result from header_filter.

Reply:
done